client/src/components/ListCover.jsx

```diff
@@ -1,59 +1,101 @@
 import {
+  Box,
   Card,
+  CardActionArea,
   CardContent,
   CardMedia,
   Grid,
   makeStyles,
+  Switch,
   Typography,
-} from "@material-ui/core"
-import React, { useContext } from "react"
-import sample from "../images/sample.jpg"
-import { userContext as context } from "../providers/UsersProvider"
+} from "@material-ui/core";
+import React, { useContext, useState } from "react";
+import { userContext as context } from "../providers/UsersProvider";
 
 const useStyles = makeStyles(() => ({
+  cardContainer: {
+    position: "relative",
+  },
   card: {
     width: 250,
     height: 350,
     backgroundColor: "white",
+    position: "relative",
   },
   image: {
-    height: 260,
+    height: 280,
   },
-}))
+}));
 
 const ListCover = ({ list }) => {
-  const classes = useStyles()
-  const userContext = useContext(context)
+  const classes = useStyles();
+  const userContext = useContext(context);
 
   const {
     updateIsListClicked,
+    axiosWithAuth,
+    openSnackbar,
     updateCurrentList,
     getListProducts,
-  } = userContext
+  } = userContext;
+
+  const { title, id, items, imageUrl, isPrivate } = list;
 
-  const { title, id, items, imageUrl } = list
+  const [toggle, setToggle] = useState(isPrivate);
 
   const clickHandler = async () => {
-    updateCurrentList(list)
-    getListProducts(id)
-    updateIsListClicked(true)
-  }
+    updateCurrentList(list);
+    getListProducts(id);
+    updateIsListClicked(true);
+  };
+
+  const toggleHandler = () => {
+    setToggle((toggle) => !toggle);
+    axiosWithAuth()
+      .put(`/lists/${id}`, { isPrivate: toggle })
```

Comment:
IMPORTANT: should be `isPrivate: !toggle`, since the state doesn't change until the component re-renders.

Reply:
yup, I guess I'll have to do another refactor as Sina suggest that I would need to move the togglehandler to the parent component. I'll fix this issue BTW.